src/main/java/net/researchgate/azkaban/GroupsLoader.java

```diff
@@ -0,0 +1,170 @@
+package net.researchgate.azkaban;
+
+import azkaban.user.Permission;
+import azkaban.user.Role;
+import org.w3c.dom.*;
+import org.xml.sax.*;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class GroupsLoader {
+
+    private static final String DEFAULT_ROLE_NAME = "admin";
+    private static final String XML_DTD_FILE_PATH = "/azkaban-groups.dtd";
+
+    Map<String, Group> loadFromFile(String filePath) {
+        File file = loadFile(filePath);
+        Document document = loadDocument(file);
+
+        if (document == null) {
+            return null;
+        }
+
+        NodeList nodes = document.getChildNodes();
+        Element rootElement = (Element)nodes.item(1);
+
+        Map<String, Role> roles = parseRoles(rootElement);
+
+        return parseGroups(rootElement, roles);
+    }
+
+    Map<String, Group> loadFromList(List<String> list) {
+        Map<String, Group> groups = new HashMap<>();
+
+        for (String groupName : list) {
+            Permission permission = new Permission();
```

Comment:
you can add this to a Constructor too ,  so you write:  new Permission(Admin);  or a factory method...
Permission.forAdmin() that returns it - same goes to other models